etl/assets.py

```diff
@@ -70,27 +66,27 @@ def wikipedia_articles_with_summaries_json_file(
 
 @asset
 def documents_of_wikipedia_articles_with_summaries(
-    wikipedia_articles_with_summaries: RecordTuple, config: OpenAiPipelineConfig
+    wikipedia_articles_with_summaries: RecordTuple,
+    openai_pipeline_config: OpenAiPipelineConfig,
 ) -> DocumentTuple:
     """Materialize an asset of Documents of Wikipedia articles with summaries."""
 
-    return DocumentTuple(
-        documents=create_documents(
-            records=wikipedia_articles_with_summaries.records,
-            record_type=config.record_type,
-            enrichment_type=config.enrichment_type,
-        )
+    return DocumentTuple.from_records(
+        records=wikipedia_articles_with_summaries.records,
+        record_type=openai_pipeline_config.record_type,
+        enrichment_type=openai_pipeline_config.enrichment_type,
     )
 
 
 @asset
 def wikipedia_articles_embeddings(
     documents_of_wikipedia_articles_with_summaries: DocumentTuple,
-    config: OpenAiSettings,
+    openai_settings: OpenAiSettings,
+    output_config: OutputConfig,
 ) -> VectorStore:
```

Comment:
Nit: pickling the VectorStore between assets is fine for a small dataset. For a larger dataset you'd want to write the VectorStore to disk, pass a path to it out of the asset, and rehydrate the VectorStore downstream. I call the "pointer" to a store that's passed between assets a "descriptor", an old file system term ("file descriptor").